Enhance docs for `Container` (#284)
mckinsey/vizro: vizro-core/docs/pages/user_guides/layouts.md

## Using the default layout
The `layout` argument of the [`Page`][vizro.models.Page] model is optional. If no layout is specified, all charts/components

> will automatically be stacked horizontally. If that is your desired layout, you can create your charts/components without providing a [`Layout`][vizro.models.Layout].

Reviewer: ```suggestion
will automatically be stacked in rows down the page. If that is your desired layout, you can create your charts/components without providing a [`Layout`][vizro.models.Layout].
```

Due to the slight ambiguity mentioned below, best to start off with non-technical language when this is first mentioned I think.

> ## Configuring the grid

Reviewer: I know it's annoying since I already approved the PR and would be good to get it merged... But I re-read this page and something I think is missing is a quick explanation of horizontal stacking. Currently we say
>  If no layout is specified, all charts/components will automatically be stacked underneath each other.

We should say also:
* this corresponds to `grid = [[0], [1], [2]]` because this specifies 3 equal rows, each of which contains a single component that takes up the whole width
* you can achieve stacking in columns with `grid = [[0, 1, 2]]` because this specifies 1 row that takes up the whole width, then divided into 3 equal columns

Maybe a short new section right above "Basic Example" called "stacking" or something would explain this?

Author: That makes sense! 💡 

![Screenshot 2024-01-30 at 19 13 29](https://github.com/mckinsey/vizro/assets/90609403/8114f396-f15e-42c6-84c6-9844523ee07b)

      [2, 2, 3, 4]]
```

> ### Stacking components

Reviewer: Are we using the terms "horizontal" and "vertical" the right way round here? To me it's kind of ambiguous which way round the definition is, because as a non-expert it sounds like horizontal stacking could mean either way round:
1. components are horizontal and stacked on top of each other (like you say here)
2. components are squashed together into one horizontal row (the opposite thing)

A quick google didn't make it clear to me which was the correct/most common definition here, so would be great if you could double check!

Author: Yeah, I see. What if we stick with the flexbox description? It's not completely ambiguous free, but then we stick with one known syntax: https://byteiota.com/flexbox-conatiner/

I could also add an image here to make it clearer and then we call it
- stacked in one row (flex-direction row means that they are stacked next to each other or vertically)
- stacked in one column (flex-direction column means when they are stacked on top of each other or horizontally)

Author: When I looked at infographics - the other way around seemed more common: https://www.c-sharpcorner.com/article/stack-and-grid-layouts-in-xamarin-forms/

**So 2) components are squashed together into one horizontal row (the opposite thing)**

Author: I am for sure no designer, but I just quickly made this in Miro and we can add it here:

![Screenshot 2024-01-31 at 10 25 38](https://github.com/mckinsey/vizro/assets/90609403/c3be66bf-172f-4f81-9ff0-846da01694df)

Reviewer: That is also the way round that dbc does it (look at the section on "Stacking objects"): https://dash-bootstrap-components.opensource.faculty.ai/docs/components/layout/. So indeed it sounds like we should do it the opposite way round from how you currently have it.

I think a couple of pictures would definitely make this clearer 😅 We already have the relevant examples in https://vizro.readthedocs.io/en/stable/pages/user_guides/layouts/?h=layout#using-custom-layout-examples (which I guess is largely there for legacy reasons, like the `one_left_two_right` stuff), so we could just copy them from that table and label them using the stacking terminology.

Author: Haha, beat you on this - see image above :) I'll add it in here then 👍

Reviewer: > I am for sure no designer, but I just quickly made this in Miro and we can add it here:
> 
> ![Screenshot 2024-01-31 at 10 25 38](https://private-user-images.githubusercontent.com/90609403/301107874-c3be66bf-172f-4f81-9ff0-846da01694df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2OTM3NzUsIm5iZiI6MTcwNjY5MzQ3NSwicGF0aCI6Ii85MDYwOTQwMy8zMDExMDc4NzQtYzNiZTY2YmYtMTcyZi00ZjgxLTlmZjAtODQ2ZGEwMTY5NGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDA5MzExNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiNGE1MDc4MzJiODFiODA2MTk0MTI4ZmFlOGJlYzE4YzJlZjZhYzFhZmZkZTBiZGRlZGNlNGJmNmZlYTk1YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4Kn3Sg4ZV_otQL2sosaW5UlYK1S9axcgXrHVYG41DjI)

This is quite beautiful I think and does the job perfectly 👌 I promote you to lead designer.

    [Grid]: ../../assets/user_guides/layout/one_left_two_right.png

### Grid - Advanced Example

> Generally, the `Layout` provides full control over the arrangement of top-level components within a Page,

Reviewer: ```suggestion
Generally, the `Layout` provides full control over the arrangement of top-level components within a page,
```

allowing arbitrarily granular control of the grid by creating larger grids.

> If you want to divide the grid into subgrids with finer control over these, you can use [Containers](container.md).

Reviewer 2: ```suggestion
If you want to divide the grid into subgrids with finer control over these, you can use [`Containers`](container.md).
```
